packages/framework-provider-kubernetes-infrastructure/src/infrastructure/k8s-sdk/k8s-management.ts

```diff
@@ -42,11 +42,12 @@ export class K8sManagement {
   public async getAllServicesInNamespace(namespace: string): Promise<Array<Service>> {
     const response = await this.unwrapResponse(this.k8sClient.listNamespacedService(namespace))
     return response.items.map((item) => {
+
       return {
         name: item.metadata?.name,
         namespace: item.metadata?.namespace ?? 'default',
         labels: item.metadata?.labels ?? {},
-        ip: item.status?.loadBalancer?.ingress?.[0]?.ip ?? '',
+        ip: item.status?.loadBalancer?.ingress?.[0]?.ip ?? item.status?.loadBalancer?.ingress?.[0]?.hostname ?? '',
```

Comment:
Is it correct that the IP is empty? If not, please add a `TODO` comment above to solve it later